<a href="https://colab.research.google.com/github/ChyzhakivskiyOlexandr/HW_5.0/blob/main/_hw_1_1_3_ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [ ]:
a = 101
b = 210
c = 111
d = 402

e = ((a + b) * c) / d
print (e)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [ ]:
a = 11283
b = 89
c = a // b
print (c)

126


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [ ]:
print (c)

126


4. Запишіть у змінну `name` ваше імʼя.

In [ ]:
name = 'Olexandr '

5. Запишіть у змінну `surname` ваше прізвище.

In [ ]:
surname = 'Chyzhakivskiy'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [ ]:
name_surname = name + surname
print (name_surname)

Olexandr Chyzhakivskiy


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [ ]:
print(name_surname.upper())

OLEXANDR CHYZHAKIVSKIY


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [ ]:
name_surname[2]

'e'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [ ]:
name_surname[2]

'e'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [ ]:
name_surname[0]


'O'

In [ ]:
name_surname[21]

'y'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [ ]:
len(surname)

13

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [ ]:
madagaskar_stolicia = 'Antananarivo'
kilkist_liter_a = madagaskar_stolicia.lower().count('a')
print(kilkist_liter_a)

4


П.С. ВИКОРИСТАВ ДОПОМОГУ ЧАТУ ДЖИПІТІ, ПРОШУ ПРОБАЧИТИ ЗА ЦЕ. У ВІДЕОУРОЦІ(А ВІН ЯК ВСІ ЗНАЮТЬ І ГООВОРЯТЬ ДУЖЕ ІНФОРМАТИВНИЙ І ПРАКТИЧНИЙ) САМЕ Я НЕ ЗНАЙШОВ ВІДПОВВІДІ НА ЦЕ ПИТАННЯ... ВИРІШИВ ЦЕ ПИТАННЯ ТАК.


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [ ]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

In [ ]:
print(my_string.startswith("The"))

True


In [ ]:
print(my_string.endswith("dog"))

False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [ ]:
print(my_string.split())

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']


In [ ]:
print('_'.join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`